lua/lspconfig/lemminx.lua

```diff
@@ -0,0 +1,51 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+local name = "lemminx"
+local bin_name = "org.eclipse.lemminx-uber.jar"
+
+configs[name] = {
+  default_config = {
+    cmd = {
+      "java", "-jar", bin_name
+    },
+    filetypes = {"xml"};
+    root_dir = function(filename)
+      return util.path.dirname(filename)
```

Comment:
Are there no other preferred project markers?

Reply:
Personally, I would like this to work for scratch-files as well, as I often create a file with an xml-response and format it using this. I tend to put my scratch-files in the /tmp-directory, so that they get deleted after a while, and are then not a part of a project structure.

This LSP does not support `textDocument/definition` yet (https://github.com/eclipse/lemminx). Correct me if I'm wrong, but I guess that is the main point of setting the project root higher up than the file location?

Or maybe it possible to use the git-root, and if that does not exists use the file location?

Reply:
You can just add the current file as a fallback, see pyright. I don't use this language server, so I'm not sure if there is a situation where having the project root set correctly matters. It is used for other features/internally by the language server.

Reply:
In maven projects with eg. multiple modules, there can be a hierarchy of pom.xml files that could reference the ones above, so I put git-root as default, and current file as fallback.

Reply:
^ `textDocument/definition` is set up to navigate between opening and closing tags in LemMinX (see https://github.com/redhat-developer/vscode-xml/blob/master/docs/Features.md#jump-to-startend-tag). LemMinX also supports type definition `textDocument/typeDefinition`, which should link from an XML element to the definition of the element in the schema file.

Reply:
ah. that's working for me. It's different to how intellij interprets "go to definition" but that's all good, I can see why lemminx would do it this way. 

... There seems to be some quirks with resolving xsd catalogs, but that's going to be an upstream lemminx thing

Thanks. Great stuff.

Reply:
good to know! added it to the doc. should be added to the lemminx readme as well I suppose though

Reply:
> good to know! added it to the doc. should be added to the lemminx readme as well I suppose though

see redhat-developer/vscode-xml#491